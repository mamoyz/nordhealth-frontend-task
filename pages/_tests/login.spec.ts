import { describe, beforeEach, it, expect, vi } from "vitest";
import { mountSuspended } from "@nuxt/test-utils/runtime";
import LoginPage from "~/pages/login.vue";
import { createI18n } from "vue-i18n";

const i18n = createI18n({
	locale: "en",
	messages: {
		en: {
			"auth.header.title": "Mocked Title",
			"auth.header.description": "Mocked Description",
			"auth.login.footer.no_account": "Mocked Footer Text 1",
			"auth.login.footer.sign_up": "Mocked Footer Text 2",
		},
		fi: {},
	},
});

const mockGlobals = {
	plugins: [i18n],
};

let wrapper: Awaited<ReturnType<typeof mountSuspended>>;

describe("Login Page", () => {
	beforeEach(async () => {
		wrapper = await mountSuspended(LoginPage, {
			global: mockGlobals,
		});
	});

	it("renders mocked header title", () => {
		expect(wrapper.text()).toContain("Mocked Title");
	});

	it("renders mocked header description", () => {
		expect(wrapper.text()).toContain("Mocked Description");
	});

	it("renders mocked footer text 1", () => {
		expect(wrapper.text()).toContain("Mocked Footer Text 1");
	});

	it("renders mocked footer text 2", () => {
		const signUpLink = wrapper.findComponent({ name: "NuxtLink" });
		expect(signUpLink.exists()).toBe(true);
		expect(signUpLink.text()).toBe("Mocked Footer Text 2");
		expect(signUpLink.props("to")).toBe("signup");
	});

	it("renders AuthForm component", () => {
		const authForm = wrapper.findComponent({ name: "AuthForm" });
		expect(authForm.exists()).toBe(true);
	});

	it("should match the snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});
});
