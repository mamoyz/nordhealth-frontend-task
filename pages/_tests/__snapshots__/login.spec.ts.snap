// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Page > should match the snapshot 1`] = `
"<div>
  <div>
    <h1 class="n-font-size-xxl n-color-accent-text">Mocked Title</h1>
    <p class="n-font-size-m n-color-text-weaker n-margin-bs-xs n-padding-i-xxl">Mocked Description</p>
  </div>
  <div class="n-margin-bs-l n-container-xs">
    <provet-stack>
      <provet-card>
        <div class="n-padding-l">
          <provet-stack>
            <provet-input data-testid="username" hide-required=""></provet-input>
            <provet-input data-testid="password" hide-required=""><template slot="end">
                <provet-button data-testid="reveal-password-button">
                  <provet-icon></provet-icon>
                </provet-button>
              </template></provet-input>
            <provet-button data-testid="submit-button"><span>auth.login.cta_title</span></provet-button>
            <!--v-if-->
          </provet-stack>
        </div>
      </provet-card>
      <div data-testid="footer" class="n-align-center n-color-text-weaker">Mocked Footer Text 1 <a href="/signup">Mocked Footer Text 2</a></div>
      <!--v-if-->
    </provet-stack>
  </div>
</div>"
`;
